platform/osgi/platform-osgi-condpermadmin/src/main/java/org/codice/ddf/condpermadmin/PermissionActivator.java

```diff
@@ -0,0 +1,205 @@
+      conditionalPermissionUpdate
+          .getConditionalPermissionInfos()
+          .add(getAllPermission(conditionalPermissionAdmin, ConditionalPermissionInfo.ALLOW));
+    } else if (priorityResult == Priority.DENY) {
+      conditionalPermissionUpdate.getConditionalPermissionInfos().addAll(allDenyInfos);
+      conditionalPermissionUpdate.getConditionalPermissionInfos().addAll(allGrantInfos);
+      conditionalPermissionUpdate
+          .getConditionalPermissionInfos()
+          .add(getAllPermission(conditionalPermissionAdmin, ConditionalPermissionInfo.DENY));
+    }
+
+    conditionalPermissionUpdate.commit();
+  }
+
+  private ConditionalPermissionInfo getAllPermission(
+      ConditionalPermissionAdmin conditionalPermissionAdmin, String type) {
+    return conditionalPermissionAdmin.newConditionalPermissionInfo(
+        null,
+        null,
+        new PermissionInfo[] {new PermissionInfo("java.security.AllPermission", "", "")},
+        type);
+  }
+
+  private void buildConditionalPermissionInfo(
+      ConditionalPermissionAdmin conditionalPermissionAdmin,
+      List<ParsedPolicyEntry> entries,
+      List<ConditionalPermissionInfo> infos,
+      String type) {
+    for (ParsedPolicyEntry parsedPolicyEntry : entries) {
+      List<ParsedPermission> permissions = parsedPolicyEntry.getPermissions();
+      PermissionInfo[] permissionInfos = new PermissionInfo[permissions.size()];
+      int index = 0;
+      for (ParsedPermission parsedPermission : permissions) {
+        permissionInfos[index++] =
+            new PermissionInfo(
+                parsedPermission.getPermissionType(),
+                replaceSystemProperties(parsedPermission.getPermissionName()),
+                parsedPermission.getActions());
+      }
+      List<ConditionInfo> conditionInfos = new ArrayList<>();
+      addCodebase(parsedPolicyEntry, conditionInfos);
+      addSignedBy(parsedPolicyEntry, conditionInfos);
+      addPrincipals(parsedPolicyEntry, conditionInfos);
+      infos.add(
+          conditionalPermissionAdmin.newConditionalPermissionInfo(
+              null,
+              (conditionInfos.isEmpty()) ? null : conditionInfos.toArray(new ConditionInfo[0]),
+              permissionInfos,
+              type));
+    }
+  }
+
+  private void addPrincipals(
+      ParsedPolicyEntry parsedPolicyEntry, List<ConditionInfo> conditionInfos) {
+    List<ParsedPrincipal> parsedPrincipals = parsedPolicyEntry.getPrincipals();
+    List<String> principals = new ArrayList<>();
+    for (ParsedPrincipal parsedPrincipal : parsedPrincipals) {
+      if (parsedPrincipal.hasAlias()) {
+        principals.add(parsedPrincipal.getAlias());
+      } else {
+        principals.add(
+            parsedPrincipal.getPrincipalClass() + "/" + parsedPrincipal.getPrincipalName());
+      }
+    }
+
+    if (!principals.isEmpty()) {
+      conditionInfos.add(new ConditionInfo(PRINCIPAL_CONDITION, principals.toArray(new String[0])));
```

Comment:
✏️  More optional to do `new String[principals.size()]` to avoid creating an array for nothing